app/tests/utils.lisp

```diff
@@ -0,0 +1,31 @@
+(in-package :qvm-app-tests)
+
+(defmacro with-mocked-function-definitions (defs &body body)
+  "Dynamically re-define global functions named in DEFS within BODY.
+
+DEFS are LET-like bindings (not FLET) whose binding keys are symbols
+that name global functions, and whose binding values are evaluated and
+return functions. These functions should have conforming lambda lists,
+types, and return values, though this is not enforced by the macro.
+
+e.g.
+(defun testfunc (a b) (+ a b))
+(with-mocked-function-definitions
+    ((testfunc (lambda (a b) (* a b))))
+  (testfunc 3 4)) ;; => 12
+"
+  (let* ((names (mapcar #'first defs))
+         (gnames (mapcar (alexandria:compose #'gensym #'symbol-name) names)))
```

Comment:
`(mapcar #'copy-symbol names)` would be fine here too.

Reply:
good to know

Reply:
cool, i didn't know about `copy-symbol`